surefire-provider/src/main/java/org/arquillian/smart/testing/surefire/provider/SmartTestingSurefireProvider.java

```diff
@@ -129,4 +169,26 @@ private boolean isGetConsoleLoggerMethod(Method method) {
             && method.getModifiers() == Modifier.PUBLIC
             && method.getReturnType() != Void.class;
     }
+
+    private void setTestMethodSelectionToParams() {
+        TestListResolver originalResolver = providerParameters.getTestRequest().getTestListResolver();
+        TestListResolver selectedResolver = new TestListResolver(selectedTestMethods);
+
+        TestListResolver newTestListResolver =
+            TestListResolver.newTestListResolver(
+                selectedResolver.getIncludedPatterns(),
+                originalResolver.getExcludedPatterns());
+
+        try {
+            Field requestedTestsField = SecurityUtils.getField(TestRequest.class, "requestedTests");
```

Comment:
I understand this is the only way to bring this functionality to our tool. I'm a bit concerned - we should be careful using reflection as this can bite us when Surefire evolves as well as when finally adding JDK 9 support.

@MatousJobanek would it make sense to at least open JIRA in Surefire project asking about a possibility to opening this API?

Reply:
Yeah JDK 9 is my nightmare.

I'll open an issue for it in the Surefire project